src/app/pages/HomePage/Graph/Graph/index.tsx

```diff
@@ -218,24 +247,46 @@ const GraphCmp: ForwardRefRenderFunction<SVGSVGElement, GraphProps> = (
   const navigate = useNavigate()
   const { isMobile } = useScreenSize()
   const [hoveredLayer, setHoveredLayer] = useState<Layer | null>(null)
-  const [outOfDate, setOutOfDate] = useState<boolean>()
+  const [outOfDateMap, setOutOfDateMap] = useState<
+    Partial<{
+      [key in Layer]: boolean | undefined
+    }>
+  >({})
 
   useEffect(() => {
     setActiveMobileGraphTooltip({ current: null })
     // should only close tooltips on isMobile change
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [isMobile])
 
-  const isLayerDisabled = (Layer: Layer) => {
-    return !RouteUtils.getEnabledLayersForNetwork(network).includes(Layer)
-  }
+  useEffect(() => {
+    setOutOfDateMap({})
+  }, [network])
 
-  const disabledMap: Record<Layer, boolean> = {
-    [Layer.emerald]: isLayerDisabled(Layer.emerald),
-    [Layer.consensus]: isLayerDisabled(Layer.consensus),
-    [Layer.cipher]: isLayerDisabled(Layer.cipher),
-    [Layer.sapphire]: isLayerDisabled(Layer.sapphire),
-  }
+  const isLayerDisabled = useCallback(
```

Comment:
Why we need to wrap `getEnabledLayersForNetwork` with useCallback when disabledMap is going to be memoized?

Reply:
Due to network dependency. Status call depends on disabledMap, which I want to trigger only when network changes.